clippy_lints/src/types.rs

```diff
@@ -68,6 +68,33 @@ declare_clippy_lint! {
     "usage of `Box<Vec<T>>`, vector elements are already on the heap"
 }
 
+/// **What it does:** Checks for use of `Vec<Box<T>>` where T: Sized anywhere in the code.
+///
+/// **Why is this bad?** `Vec` already keeps its contents in a separate area on
+/// the heap. So if you `Box` its contents, you just add another level of indirection.
+///
+/// **Known problems:** Vec<Box<T: Sized>> makes sense if T is a large type (see #3530, 1st comment).
+///
+/// **Example:**
+/// ```rust
+/// struct X {
+///     values: Vec<Box<i32>>,
+/// }
+/// ```
+///
+/// Better:
+///
+/// ```rust
+/// struct X {
+///     values: Vec<i32>,
+/// }
+/// ```
+declare_clippy_lint! {
+    pub VEC_BOX_SIZED,
+    complexity,
```

Comment:
can it be `perf`? cc @oli-obk

Reply:
@flip1995 tagged the original issue as complexity so that's what I went with.

Reply:
can be either. Most of our type complexity lints are `complexity` even though they can have perf effects

Reply:
I'm sticking with `complexity`, unless there's a compelling reason not to, feels more right to me.

Reply:
works for me